bumble/hid.py

```diff
@@ -242,37 +282,217 @@ def on_l2cap_channel_open(self, l2cap_channel: l2cap.ClassicChannel) -> None:
             self.l2cap_intr_channel.sink = self.on_intr_pdu
         logger.debug(f'$$$ L2CAP channel open: {l2cap_channel}')
 
+    def on_l2cap_channel_close(self, l2cap_channel: l2cap.ClassicChannel) -> None:
+        if l2cap_channel.psm == HID_CONTROL_PSM:
+            self.l2cap_ctrl_channel = None
+        else:
+            self.l2cap_intr_channel = None
+        logger.debug(f'$$$ L2CAP channel close: {l2cap_channel}')
+
     def on_ctrl_pdu(self, pdu: bytes) -> None:
         logger.debug(f'<<< HID CONTROL PDU: {pdu.hex()}')
-        # Here we will receive all kinds of packets, parse and then call respective callbacks
-        message_type = pdu[0] >> 4
         param = pdu[0] & 0x0F
+        message_type = pdu[0] >> 4
 
         if message_type == Message.MessageType.HANDSHAKE:
             logger.debug(f'<<< HID HANDSHAKE: {Message.Handshake(param).name}')
             self.emit('handshake', Message.Handshake(param))
+        elif message_type == Message.MessageType.GET_REPORT:
+            logger.debug('<<< HID GET REPORT')
+            self.handle_get_report(pdu)  #  type: ignore[attr-defined]
```

Comment:
I think `handle_get_report()` belongs to `Device` class instead of `HID` class?

Reply:
Yes "handle_" belong to Device class, and "on_ctrl_pdu" belongs to HID Class. So to avoid mypy error we have defined them in parent class(HID) too. Please let us know if this is right approach.

def on_ctrl_pdu(self, pdu: bytes) -> None: